azure-pipelines.yml

```diff
@@ -169,112 +185,225 @@ jobs:
     inputs:
       command: test
-      projects: '**/Microsoft.Spark.E2ETest/*.csproj'
+      projects: 'master/**/Microsoft.Spark.E2ETest/*.csproj'
       arguments: '--configuration $(buildConfiguration) --filter $(TestsToFilterOut)'
     env:
       SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.0-bin-hadoop2.7
       DOTNET_WORKER_DIR: $(OldestCompatibleDotnetWorkerDir)
 
   - task: DotNetCoreCLI@2
-    displayName: 'E2E tests for Spark 2.4.1 with oldest compatible worker $(oldestCompatibleWorker)'
+    displayName: 'E2E tests for Spark 2.4.1 with oldest compatible worker v$(backwardCompatibleRelease)'
     inputs:
       command: test
-      projects: '**/Microsoft.Spark.E2ETest/*.csproj'
+      projects: 'master/**/Microsoft.Spark.E2ETest/*.csproj'
       arguments: '--configuration $(buildConfiguration) --filter $(TestsToFilterOut)'
     env:
       SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.1-bin-hadoop2.7
       DOTNET_WORKER_DIR: $(OldestCompatibleDotnetWorkerDir)
   
   - task: DotNetCoreCLI@2
-    displayName: 'E2E tests for Spark 2.4.3 with oldest compatible worker $(oldestCompatibleWorker)'
+    displayName: 'E2E tests for Spark 2.4.3 with oldest compatible worker v$(backwardCompatibleRelease)'
     inputs:
       command: test
-      projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
+      projects: 'master/**/Microsoft.Spark*.E2ETest/*.csproj'
       arguments: '--configuration $(buildConfiguration) --filter $(TestsToFilterOut)'
     env:
       SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.3-bin-hadoop2.7
       DOTNET_WORKER_DIR: $(OldestCompatibleDotnetWorkerDir)
 
   - task: DotNetCoreCLI@2
-    displayName: 'E2E tests for Spark 2.4.4 with oldest compatible worker $(oldestCompatibleWorker)'
+    displayName: 'E2E tests for Spark 2.4.4 with oldest compatible worker v$(backwardCompatibleRelease)'
     inputs:
       command: test
-      projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
+      projects: 'master/**/Microsoft.Spark*.E2ETest/*.csproj'
       arguments: '--configuration $(buildConfiguration) --filter $(TestsToFilterOut)'
     env:
       SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.4-bin-hadoop2.7
       DOTNET_WORKER_DIR: $(OldestCompatibleDotnetWorkerDir)
 
   - task: DotNetCoreCLI@2
-    displayName: 'E2E tests for Spark 2.4.5 with oldest compatible worker $(oldestCompatibleWorker)'
+    displayName: 'E2E tests for Spark 2.4.5 with oldest compatible worker v$(backwardCompatibleRelease)'
     inputs:
       command: test
-      projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
+      projects: 'master/**/Microsoft.Spark*.E2ETest/*.csproj'
       arguments: '--configuration $(buildConfiguration) --filter $(TestsToFilterOut)'
     env:
       SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.5-bin-hadoop2.7
       DOTNET_WORKER_DIR: $(OldestCompatibleDotnetWorkerDir)
 
+  - task: Maven@3
+    displayName: 'Maven build src for oldest compatible release v$(forwardCompatibleRelease)'
```

Comment:
forward?

Reply:
Do we also need to change the display name for each e2e test with forward and backward?

Reply:
yes, let's do that and stick with backward compatible and forward compatible.